src/main/java/com/marklogic/kafka/connect/sink/SinkRecordConverter.java

```diff
@@ -10,6 +11,6 @@
  */
 public interface SinkRecordConverter {
 
-	DocumentWriteOperation convert(SinkRecord sinkRecord);
+	DocumentWriteOperation convert(SinkRecord sinkRecord)  throws IOException;
```

Comment:
What's the benefit of throwing an IOException from this? Can the client possibly do anything with it, other than catching it and logging it? And is there a reason the client should catch and log an IOException but not do that for any other Exception type?

Reply:
Clean this up in the follow-up refactor issue.